index.php

```diff
@@ -315,7 +323,38 @@ function blockbase_create_theme_menu() {
 function blockbase_save_theme() {
 	// I can't work out how to call the API but this works for now.
 	if ( ! empty( $_GET['page'] ) && $_GET['page'] === 'create-blockbase-theme' && ! empty( $_GET['theme'] ) ) {
+
+		// Check user capabilities.
+		if ( ! current_user_can( 'edit_theme_options' ) ) {
+			return add_action( 'admin_notices', 'create_blockbase_child_admin_notice_error' );
+		}
+
+		// Check nonce
+		if ( ! wp_verify_nonce( $_GET['nonce'], 'create_blockbase_theme' ) ) {
+			return add_action( 'admin_notices', 'create_blockbase_child_admin_notice_error' );
+		}
+
+		if ( empty( $_GET['theme']['name'] ) ) {
+			return add_action( 'admin_notices', 'create_blockbase_child_admin_notice_error' );
+		}
+
+		return add_action( 'admin_notices', 'create_blockbase_child_admin_notice_success' );
 		gutenberg_edit_site_export_theme( $_GET['theme'] );
 	}
 }
 add_action( 'admin_init', 'blockbase_save_theme');
+
+function create_blockbase_child_admin_notice_error() {
+	$class = 'notice notice-error';
+	$message = __( 'Please specify a theme name.', 'create-blockbase-theme' );
```

Comment:
Sometimes you can get in here for other errors, but only ones that users shouldn't hit unless they are doing something dodgy!